docs/sources/flow/reference/components/otelcol.processor.resourcedetection.md

```diff
@@ -0,0 +1,940 @@
+
+### aks
+
+The `aks` block adds resource attributes related to Azure AKS.
+
+The `aks` block supports the following blocks:
+
+Block                                          | Description                                       | Required
+---------------------------------------------- | ------------------------------------------------- | --------
+[resource_attributes](#aks--resource_attributes) | Configures which resource attributes to add.      | no
+
+#### aks > resource_attributes
+
+The `resource_attributes` block supports the following blocks:
+
+Block                           | Description                                                                                 | Required
+------------------------------- | ------------------------------------------------------------------------------------------- | --------
+[cloud.platform][res-attr-cfg]  | Toggles the `cloud.platform` resource attribute. <br> Sets `enabled` to `true` by default.  | no
+[cloud.provider][res-attr-cfg]  | Toggles the `cloud.provider` resource attribute. <br> Sets `enabled` to `true` by default.  | no
+
+Example values:
+* `cloud.provider`: `"azure"`
+* `cloud.platform`: `"azure_vm"`
+
+### consul
+
+The `consul` block queries a Consul agent and reads its configuration endpoint to retrieve values for resource attributes.
+
+The `consul` block supports the following attributes:
+
+Attribute    | Type           | Description                                                                 | Default     | Required
+------------ |--------------- | --------------------------------------------------------------------------- |-------------| --------
+`address`    | `string`       | The address of the Consul server | `""`        | no
+`datacenter` | `string`       | Datacenter to use. If not provided, the default agent datacenter is used. | `""`        | no
+`token`      | `secret`       | A per-request ACL token which overrides the Consul agent's default (empty) token. | `""`        | no
+`namespace`  | `string`       | The name of the namespace to send along for the request. | `""`        | no
+`meta`       | `list(string)` | Allowlist of [Consul Metadata][] keys to use as resource attributes. | `[]`        | no
+
+`token` is only required if [Consul's ACL System][] is enabled.
+
+[Consul Metadata]: https://www.consul.io/docs/agent/options#node_meta
+[Consul's ACL System]: https://www.consul.io/docs/security/acl/acl-system
+
+The `consul` block supports the following blocks:
+
+Block                                          | Description                                       | Required
+---------------------------------------------- | ------------------------------------------------- | --------
+[resource_attributes](#consul--resource_attributes) | Configures which resource attributes to add.      | no
+
+#### consul > resource_attributes
+
+The `resource_attributes` block supports the following blocks:
+
+<!-- TODO: Why are all these azure resource attributes here? -->
```

Comment:
I need to check this too:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.87.0/processor/resourcedetectionprocessor/internal/consul/internal/metadata/generated_config.go#L28-L37

Reply:
I raised a PR to fix this upstream:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30727